pages/stories/[id].js

```diff
@@ -1,10 +1,19 @@
 import { useRouter } from "next/router";
 import StoryDetailCard from "@/components/Story";
-import stories from "@/public/db";
+import { useState, useEffect } from "react";
 
 function StoryDetailPage() {
+  const [stories, setStories] = useState([]);
+
   const router = useRouter();
   const { id } = router.query;
+  useEffect(() => {
+    const storedStories = JSON.parse(localStorage.getItem("stories"));
+
+    if (storedStories && Array.isArray(storedStories)) {
```

Comment:
If I understand correctly, you want to set stories to storedStories `if` storedStories is true `&&` storedStories is an array. Since you're initializing your localStorage 'stories' item as an array, and an empty array is truthy, this `if` will always resolve to true. Maybe you want to check if the `storedStories.length > 0` ? This way the code only executes if there's at least one story inside the stories array from localStorage.

But that's just my two cents. Feel free to ask a coach about this.

Reply:
i think u r absolutely right. thx ! i also think i will continue by switching the place where i store my data. using mongoose might be better choice since localstorage is small and images stack up with every every new story